### DISCUSSION TITLE
Get-PnPSiteTemplate - v1.6.0 - Are templates backwards compatible?

### DISCUSSION THREAD

**Discussion by Nibushi** [pnp/pnpframework#393] [Q&A] [answered]
I've been doing some testing with creating PnP Templates using PnP.PowerShell v1.6.0. If I create and invoke these templates using v1.6.0 then everything works as expected. However, I have some Azure Functions which are using PnP.PowerShell v1.4.0 and the v1.6.0 templates don't work. 

With further manual testing I notice that these v1.6.0 templates don't work with any previous version of PnP.PowerShell. Is that to be expected? If that is the expected behaviour, is it due to this line in the template which specifies the PnP.Framework version to use:
<pnp:Preferences Generator="PnP.Framework, Version=1.5.0.0, Culture=neutral, PublicKeyToken=null" />

Thanks
Simon

**Comment by PedroMordeP** [comment, marked as answer]
Hello, I don't think so,

So if you collect a template using a newer version and try to deploy it with an older version you should check if that version supports the same template schema.

To check the template schema, open the file and check this property xmlns:pnp="http://schemas.dev.office.com/PnP/2021/03/ProvisioningSchema"

The PnP/2021/03 is the schema used.

You can see on the following link all the schemas available https://github.com/pnp/PnP-Provisioning-Schema/tree/master/Samples

**Reply by Nibushi** [reply]
Thanks for the info. Apologies for the delay in replying.
